mesheryctl/internal/cli/root/registry/update.go

```diff
@@ -262,13 +322,116 @@ func logModelUpdateSummary(modelToCompUpdateTracker *store.GenerticThreadSafeSto
 	utils.Log.Info(fmt.Sprintf("For %d models updated %d components", len(values), totalAggregateComponents))
 }
 
+func parseLocalCSVDirectory(dirPath string) (map[string]map[string][]utils.ComponentCSV, error) {
+	localComps := make(map[string]map[string][]utils.ComponentCSV)
+	var allErrors []error
+
+	info, err := os.Stat(dirPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not access csv directory '%s': %w", dirPath, err)
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("provided path is not a directory: %s", dirPath)
+	}
+
+	entries, err := os.ReadDir(dirPath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			// optionally recurse or skip
+			continue
+		}
+		if filepath.Ext(entry.Name()) == ".csv" {
+			csvPath := filepath.Join(dirPath, entry.Name())
+			fileComps, err := parseOneCSVFile(csvPath)
+			if err != nil {
+				allErrors = append(allErrors, err)
+				continue
+			}
+			// Merge fileComps into localComps
+			for registrant, modelMap := range fileComps {
+				if _, ok := localComps[registrant]; !ok {
+					localComps[registrant] = make(map[string][]utils.ComponentCSV)
+				}
+				for modelName, comps := range modelMap {
+					localComps[registrant][modelName] = append(localComps[registrant][modelName], comps...)
+				}
+			}
+		}
+	}
+	if len(allErrors) > 0 {
+		return nil, errors.Join(allErrors...)
+	}
+	return localComps, nil
+}
+
+// parseOneCSVFile reads a single CSV file, enforces the required columns, and returns
+// a nested map structure: map[registrant]map[modelName][]utils.ComponentInfo
+func parseOneCSVFile(csvPath string) (map[string]map[string][]utils.ComponentCSV, error) {
```

Comment:
Since we can use the same parsing function for google sheet, we can remove this one (like I commented above).